packages/database/src/core/util/libs/parser.ts

```diff
@@ -37,6 +37,25 @@ function decodePath(pathString: string): string {
   return pathStringDecoded;
 }
 
+/**
+ * @param {!string} queryString
+ * @return {!Map<string, string>} key value hash
+ */
+function decodeQuery(queryString: string): Map<string, string> {
+  let results = new Map<string, string>();
+  const pieces = queryString.split('&');
+  for (let i = 0; i < pieces.length; i++) {
+    if (pieces[i].length > 0) {
+      try {
+        const piece = decodeURIComponent(pieces[i]);
+        const kv = piece.split('=');
+        results.set(kv[0], kv[1]);
```

Comment:
Don't you need to split first and then decode?

Reply:
yep